src/test/java/com/google/sps/servlets/ShowInterviewersServletTest.java

```diff
@@ -0,0 +1,370 @@
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.sps.data.Availability;
+import com.google.sps.data.FakeAvailabilityDao;
+import com.google.sps.data.FakePersonDao;
+import com.google.sps.data.Person;
+import com.google.sps.data.PossibleInterviewer;
+import com.google.sps.data.TimeRange;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.junit.Test;
+import com.google.gson.JsonSyntaxException;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.servlet.ServletException;
+
+@RunWith(JUnit4.class)
+public final class ShowInterviewersServletTest {
+  LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalUserServiceTestConfig());
+  private FakeAvailabilityDao availabilityDao;
+  private FakePersonDao personDao;
+
+  private final String person1Email = "person1@mail";
+  private final String person1Id = String.format("%d", person1Email.hashCode());
+  private final Person person1 =
+      Person.create(person1Id, person1Email, "Test", "Subject", "Google", "SWE", "linkedIn");
+
+  private final String person2Email = "person2@mail";
+  private final String person2Id = String.format("%d", person2Email.hashCode());
+  private final Person person2 =
+      Person.create(person2Id, person2Email, "Test", "Subject", "Google", "PM", "linkedIn");
+
+  private final String person3Email = "person3@mail";
+  private final String person3Id = String.format("%d", person3Email.hashCode());
+  private final Person person3 =
+      Person.create(person3Id, person3Email, "Test", "Subject", "Google", "PM", "linkedInLink");
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    availabilityDao = new FakeAvailabilityDao();
+    personDao = new FakePersonDao();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void onlyReturnsInterviewersWhoAreAvailableForTheFullHour() throws IOException {
+    personDao.create(person1);
+    ShowInterviewersServlet servlet = new ShowInterviewersServlet();
+    servlet.init(availabilityDao, personDao);
+    helper.setEnvIsLoggedIn(true).setEnvEmail("person@gmail.com").setEnvAuthDomain("auth");
+
+    availabilityDao.create(
```

Comment:
I see a lot of availabilities in the tests that use the same values. Would it be better to initialize them similar to the way you initialized values for users?

Reply:
Done